lstchain/scripts/lstchain_merge_hdf5_files.py

```diff
@@ -28,13 +28,22 @@
                     dest='noimage',
                     help='Boolean. True to remove the images',
                     default=False)
-
-args = parser.parse_args()
+                                                                                    
+parser.add_argument('--run-number', '-r', action='store', type=str,                                                           
+                    dest='run_number',                                                                                  
+                    help='Merge files run-wise',                                                                        
+                    default=None)                                                                                       
```

Comment:
you could also set `default=''` and avoid a condition check (`'' in f` is always True) but it's a matter of preference here

Reply:
Maybe `None` is more clear to the user. I would just add a comment about default in the help anyway. "Merge files run-wise if a run number is passed, otherwise merge all files in the directory".

Reply:
Done